holoviews/core/spaces.py

```diff
@@ -881,11 +881,12 @@ class DynamicMap(HoloMap):
         If the callable is an instance of Callable it will be used
         directly, otherwise it will be automatically wrapped in one.""")
 
-    streams = param.List(default=[], constant=True, doc="""
-       List of Stream instances to associate with the DynamicMap. The
-       set of parameter values across these streams will be supplied as
-       keyword arguments to the callback when the events are received,
-       updating the streams.""" )
+    streams = param.ClassSelector(default=[], class_=(dict, list), constant=True, doc="""
```

Comment:
```suggestion
    streams = param.List(default=[], constant=True, doc="""
```

Reply:
@jbednar I would prefer to give more explicit indication that `streams=` can be supplied as a dict or a list using `ClassSelector` but Philipp is also right that after the constructor it will always be a list. Happy to let you be the tie breaker!

Reply:
I'll also note that `constant=True` has always been declared so you were never meant to set `streams` after the constructor has run.

Reply:
Good question. I guess `param.List` correctly reports what the `streams` parameter holds, so I agree with this suggestion. It's just important to document in the constructor's docstring that dict is also accepted.